plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -467,9 +467,10 @@ func (s *Plugin) mountServiceAccountToken(serviceAccount *corev1.ServiceAccount,
 			tokenVolumeName = s.generateName(ServiceAccountVolumeName + "-")
 		} else {
 			// Try naming the volume the same as the serviceAccountToken, and uniquify if needed
-			tokenVolumeName = serviceAccountToken
+			// Replace dots because volumeMountName can't contain it
+			tokenVolumeName = strings.Replace(serviceAccountToken, ".", "-", -1)
```

Comment:
are token secrets whose name is > 63 chars also an issue?

Reply:
No, there is no problem with the token name length. For example:

Token for that service account
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  namespace: default
  name: this-is-longest-service-account-name-in-the-world-to-test-that-names-longer-than-sixty-three-characters-dont-work-for-volume-mount
```
is `this-is-longest-service-account-name-in-the-world-to-test-v5lcp` (63 chars)

And so the volume looks like:
```yaml
  volumes:
  - name: this-is-longest-service-account-name-in-the-world-to-test-v5lcp
    secret:
      defaultMode: 420
      secretName: this-is-longest-service-account-name-in-the-world-to-test-v5lcp
```

Even if volume with that name already exists, `generatedName` function do the job:
```yaml
  volumes:
  - emptyDir: {}
    name: this-is-longest-service-account-name-in-the-world-to-test-v5lcp
  - name: this-is-longest-service-account-name-in-the-world-to-test-fk56m
    secret:
      defaultMode: 420
      secretName: this-is-longest-service-account-name-in-the-world-to-test-v5lcp
```

Reply:
thanks for checking